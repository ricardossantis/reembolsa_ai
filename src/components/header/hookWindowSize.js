import { useState, useEffect } from "react";

export const useWindowSize = () => {
  const initialState = {
    width: 0,
    height: 0,
  };

  const [windowSize, setWindowSize] = useState(initialState);

  useEffect(() => {
    const handleSize = () => {
      setWindowSize({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    };
    window.addEventListener("resize", handleSize);
    handleSize();
    return () => window.removeEventListener("resize", handleSize);
  }, []);
  return windowSize;
};
