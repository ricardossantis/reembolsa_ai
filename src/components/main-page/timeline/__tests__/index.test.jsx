import React from "react";
import renderer from "react-test-renderer";
import MainTimeline from "../index";
import { useWindowSize } from "../../../system-general/header/hookWindowSize.js";

const timelineItem = {
  item1: "test1",
  item2: "test2",
  item3: "test3",
  item4: "test4",
  item5: "test5",
  item6: "test6",
  item7: "test7",
  item8: "test8",
  item9: "test9",
  item10: "test10",
};

jest.mock("../../../system-general/header/hookWindowSize.js", () => ({
  useWindowSize: jest.fn(),
}));

it("width < 768", () => {
  useWindowSize.mockImplementation(() => {
    return { width: 767 };
  });
  const tree = renderer.create(<MainTimeline {...timelineItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("width > 768", () => {
  useWindowSize.mockImplementation(() => {
    return { width: 769 };
  });
  const tree = renderer.create(<MainTimeline {...timelineItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});
