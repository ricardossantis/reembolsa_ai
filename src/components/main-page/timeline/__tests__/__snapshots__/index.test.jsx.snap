// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`width < 768 1`] = `
<div>
  <div
    className="sc-gtssRu kWnWji"
  >
    <ul
      className="ant-timeline"
    >
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test1
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test2
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test3
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test4
        </div>
      </li>
      <li
        className="ant-timeline-item ant-timeline-item-last"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test5
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`width > 768 1`] = `
<div>
  <div
    className="sc-bdnylx grBsSd"
  >
    <ul
      className="ant-timeline"
    >
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test1
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test2
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test3
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test4
        </div>
      </li>
      <li
        className="ant-timeline-item ant-timeline-item-last"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test5
        </div>
      </li>
    </ul>
    <ul
      className="ant-timeline"
    >
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test6
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test7
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test8
        </div>
      </li>
      <li
        className="ant-timeline-item"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test9
        </div>
      </li>
      <li
        className="ant-timeline-item ant-timeline-item-last"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-#2CBFD3"
          style={
            Object {
              "borderColor": "#2CBFD3",
            }
          }
        />
        <div
          className="ant-timeline-item-content"
        >
          test10
        </div>
      </li>
    </ul>
  </div>
</div>
`;
